### ISSUE TITLE
Add `random_state` hyperparameter to make results reproducible

### ISSUE THREAD

**Issue by x-tabdeveloping** [sashakolpakov/dire-jax#27]
Most estimators that rely on stochasticity in sklearn have a `random_state` hyperparameter.
As far as I understand, your model uses random numbers both for initialization and sampling.
I believe being able to specify a random seed this way would be a substantial improvement.

**Comment by sashakolpakov** [comment]
Addressed in https://github.com/sashakolpakov/dire-jax/pull/28 by @x-tabdeveloping
